### ISSUE TITLE
[BUG]: 

### ISSUE THREAD

**Issue by kakalovedata** [pycaret/pycaret#3357]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

>>> from pycaret.classification import *

>>> 
>>> best_model=load_model('pangu_model_03')
Transformation Pipeline and Model Successfully Loaded
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/classification/functional.py", line 2355, in load_model
    return experiment.load_model(
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/classification/oop.py", line 2962, in load_model
    return super().load_model(
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/internal/pycaret_experiment/tabular_experiment.py", line 2480, in load_model
    return pycaret.internal.persistence.load_model(
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/internal/persistence.py", line 399, in load_model
    return joblib.load(model_name)
  File "/root/anaconda3/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 658, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/root/anaconda3/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
    obj = unpickler.load()
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1212, in load
    dispatch[key[0]](self)
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1537, in load_stack_global
    self.append(self.find_class(module, name))
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1579, in find_class
    __import__(module, level=0)
ModuleNotFoundError: No module named 'pycaret.internal.memory'

### Reproducible Example

```python
>>> from pycaret.classification import *


>>> 
>>> best_model=load_model('pangu_model_03')
Transformation Pipeline and Model Successfully Loaded
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/classification/functional.py", line 2355, in load_model
    return experiment.load_model(
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/classification/oop.py", line 2962, in load_model
    return super().load_model(
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/internal/pycaret_experiment/tabular_experiment.py", line 2480, in load_model
    return pycaret.internal.persistence.load_model(
  File "/root/anaconda3/lib/python3.9/site-packages/pycaret/internal/persistence.py", line 399, in load_model
    return joblib.load(model_name)
  File "/root/anaconda3/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 658, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/root/anaconda3/lib/python3.9/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
    obj = unpickler.load()
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1212, in load
    dispatch[key[0]](self)
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1537, in load_stack_global
    self.append(self.find_class(module, name))
  File "/root/anaconda3/lib/python3.9/pickle.py", line 1579, in find_class
    __import__(module, level=0)
ModuleNotFoundError: No module named 'pycaret.internal.memory'
```

### Expected Behavior

ModuleNotFoundError: No module named 'pycaret.internal.memory'

linux Ubuntu  18.04 64

dont kown why

### Actual Results

```python-traceback
1
```

### Installed Versions

<details>
Replace this line with the output of the version code above.
</details>
1

**Comment by ngupta23** [comment]
We have been making backend changes to the code. The above error comes when you created the model using a certain version of pycaret and then loaded it later in a different version of pycaret.

In order to avoid these issues, the saved model must be loaded in the same version of pycaret as the one used to create it. Hope it helps.

**Comment by kakalovedata** [comment]
thank u very much

**Comment by kakalovedata** [comment]
i try but not work:

on my mac : 

Name: pycaret
Version: 3.0.0rc8
Summary: PyCaret - An open source, low-code machine learning library in Python.
Home-page: https://github.com/pycaret/pycaret
Author: Moez Ali
Author-email: moez.ali@queensu.ca
License: MIT
Location: /Users/master/miniconda3/lib/python3.8/site-packages
Requires: category-encoders, cloudpickle, deprecation, imbalanced-learn, importlib-metadata, ipython, ipywidgets, jinja2, joblib, kaleido, lightgbm, markupsafe, matplotlib, nbformat, numba, numpy, pandas, plotly, plotly-resampler, pmdarima, psutil, pyod, requests, schemdraw, scikit-learn, scikit-plot, scipy, sktime, statsmodels, tbats, tqdm, xxhash, yellowbrick
Required-by: 



my linux:
(base) root@iZ8vb6bluim1i2i3b75l43Z:~# pip show pycaret
Name: pycaret
Version: 3.0.0rc8
Summary: PyCaret - An open source, low-code machine learning library in Python.
Home-page: https://github.com/pycaret/pycaret
Author: Moez Ali
Author-email: moez.ali@queensu.ca
License: MIT
Location: /root/miniconda3/lib/python3.8/site-packages
Requires: category-encoders, cloudpickle, deprecation, imbalanced-learn, importlib-metad

**Comment by ngupta23** [comment]
We can not reproduce this example with the information that you have provided. Can you give a complete reproducible example from start to finish (how model was created and saved, then how it was loaded and used). Also document the versions used in each step using the show_versions method. Thanks!

**Comment by biased-ai** [comment]
Hi I am also facing the same issue....

My model has been trained on pycaret 2.3.10, and model are also being loaded there only... But when I use predict_model in a new console, it returns that estimator has no attribute decision_function

**Comment by biased-ai** [comment]
predictions=predict_model(model_file, data=score_df) 
Traceback (most recent call last):

  File "<ipython-input-8-e1aa64ae3cb9>", line 1, in <module>
    predictions=predict_model(model_file, data=score_df)

  File "/usr/local/lib/python3.6/site-packages/pycaret/anomaly.py", line 873, in predict_model
    return experiment.predict_model(estimator=model, data=data)

  File "/usr/local/lib/python3.6/site-packages/pycaret/internal/pycaret_experiment/unsupervised_experiment.py", line 913, in predict_model
    pred_score = estimator.decision_function(data_transformed)

  File "/usr/local/lib/python3.6/site-packages/sklearn/utils/metaestimators.py", line 113, in __get__
    getattr(delegate, self.attribute_name)

AttributeError: 'HuberRegressor' object has no attribute 'decision_function'

**Comment by ngupta23** [comment]
```
File "/usr/local/lib/python3.6/site-packages/pycaret/anomaly.py", line 873, in predict_model
```
```
AttributeError: 'HuberRegressor' object has no attribute 'decision_function'
```

Why are you calling a `HuberRegressor` model using the Anomaly detection module?

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
